moonraker/components/wled.py

```diff
@@ -217,6 +202,72 @@ async def set_wled(self: Strip,
             # next transmitting
             self.send_full_chain_data = True
 
+class StripHttp(Strip):
+    def __init__(self: StripHttp,
+                 name: str,
+                 color_order: ColorOrder,
+                 cfg: ConfigHelper):
+        super().__init__(name, color_order, cfg)
+
+        # Read the uri information
+        addr: str = cfg.get("address")
+        port: int = cfg.getint("port", 80)
+        protocol: str = cfg.get("protocol", "http")
+        self.url = f"{protocol}://{addr}:{port}/json"
+
+        self.timeout: float = cfg.getfloat("timeout", 2.)
+        self.client = AsyncHTTPClient()
+
+    async def send_wled_command_impl(self: StripHttp,
+                                     state: Dict[str, Any]) -> None:
+        async with self.request_mutex:
+            logging.debug(f"WLED: url:{self.url} json:{state}")
+
+            headers = {"Content-Type": "application/json"}
+            request = HTTPRequest(url=self.url,
+                                  method="POST",
+                                  headers=headers,
+                                  body=json.dumps(state),
+                                  connect_timeout=self.timeout,
+                                  request_timeout=self.timeout)
+            response = await self.client.fetch(request)
+
+            logging.debug(
+                f"WLED: url:{self.url} status:{response.code} "
+                f"response:{response.body}")
+
+class StripSerial(Strip):
+    def __init__(self: StripSerial,
+                 name: str,
+                 color_order: ColorOrder,
+                 cfg: ConfigHelper):
+        super().__init__(name, color_order, cfg)
+
+        # Read the serial information (requires wled 0.13 2108250 or greater)
+        serialport: str = cfg.get("serial")
+        baud: int = cfg.getint("baud", 115200, above=49)
+
+        # write_timeout of 0 is non-blocking
+        self.ser = serial.Serial(serialport, baud,
+                                 write_timeout=0)
```

Comment:
After the serial port is opened it is good practice to use `os.setblocking(fd, False)` to make sure that underlying file will not block.  In this case the fd is returned by `self.ser.fileno()`.

Reply:
Made the suggested changes